fastapi/security/jwt.py

```diff
@@ -0,0 +1,444 @@
+        )
+        return jwt_encoded
+
+    def create_refresh_token(
+        self, subject: Dict[str, Any], expires_delta: Optional[timedelta] = None
+    ) -> str:
+        expires_delta = (
+            expires_delta if expires_delta is not None else self.refresh_expires_delta
+        )
+        to_encode = self._generate_payload(subject, expires_delta)
+
+        # Adding creating refresh token mark
+        to_encode["type"] = "refresh"
+
+        jwt_encoded: str = jwt.encode(
+            to_encode, self.secret_key, algorithm=self.algorithm
+        )
+        return jwt_encoded
+
+    @staticmethod
+    def set_access_cookie(
+        response: Response, access_token: str, expires_delta: Optional[timedelta] = None
+    ) -> None:
+        seconds_expires: Optional[int] = (
+            int(expires_delta.total_seconds()) if expires_delta else None
+        )
+        response.set_cookie(
+            key="access_token_cookie",
+            value=access_token,
+            httponly=False,
+            max_age=seconds_expires,  # type: ignore
+        )
+
+    @staticmethod
+    def set_refresh_cookie(
+        response: Response,
+        refresh_token: str,
+        expires_delta: Optional[timedelta] = None,
+    ) -> None:
+        seconds_expires: Optional[int] = (
+            int(expires_delta.total_seconds()) if expires_delta else None
+        )
+        response.set_cookie(
+            key="refresh_token_cookie",
+            value=refresh_token,
+            httponly=True,
+            max_age=seconds_expires,  # type: ignore
+        )
+
+    @staticmethod
+    def unset_access_cookie(response: Response) -> None:
+        response.set_cookie(
+            key="access_token_cookie", value="", httponly=False, max_age=-1
+        )
+
+    @staticmethod
+    def unset_refresh_cookie(response: Response) -> None:
+        response.set_cookie(
+            key="refresh_token_cookie", value="", httponly=True, max_age=-1
+        )
+
+
+class JwtAccess(JwtAuthBase):
+    _bearer = JwtAuthBase.JwtAccessBearer()
+    _cookie = JwtAuthBase.JwtAccessCookie()
+
+    def __init__(
+        self,
+        secret_key: str,
+        places: Optional[Set[str]] = None,
+        auto_error: bool = True,
+        algorithm: str = jwt.ALGORITHMS.HS256,
+        access_expires_delta: Optional[timedelta] = None,
+        refresh_expires_delta: Optional[timedelta] = None,
+    ):
+        super().__init__(
+            secret_key,
+            places=places,
+            auto_error=auto_error,
+            algorithm=algorithm,
+            access_expires_delta=access_expires_delta,
+            refresh_expires_delta=refresh_expires_delta,
+        )
+
+    def _get_credentials(
+        self,
+        bearer: Optional[JwtAuthBase.JwtAccessBearer],
+        cookie: Optional[JwtAuthBase.JwtAccessCookie],
+    ) -> Optional[JwtAuthorizationCredentials]:
+        payload = self._get_payload(bearer, cookie)
+
+        if payload:
```

Comment:
This is where you reverse the condition the success case should be at the bottom. [It is called early return or fast fail.](https://szymonkrajewski.pl/why-should-you-return-early/)